milli/src/update/index_documents/helpers/grenad_helpers.rs

```diff
@@ -64,16 +64,21 @@ pub fn sorter_into_reader(
     writer_into_reader(writer)
 }
 
-pub fn writer_into_reader(writer: grenad::Writer<File>) -> Result<grenad::Reader<File>> {
-    let mut file = writer.into_inner()?;
+pub fn writer_into_reader(
+    writer: grenad::Writer<BufWriter<File>>,
+) -> Result<grenad::Reader<BufReader<File>>> {
+    let mut file = writer
+        .into_inner()?
+        .into_inner()
+        .map_err(|err| InternalError::BufIntoInnerError(err.to_string()))?;
```

Comment:
Can't we use the [`IntoInnerError::into_error`](https://doc.rust-lang.org/std/io/struct.IntoInnerError.html#method.into_error) method to return a classic `io::Error` instead?

Reply:
Oh yes nice, didn't know it existed!